src/lib/sql/sql_pipeline_statement.cpp

```diff
@@ -206,8 +208,9 @@ const std::shared_ptr<AbstractOperator>& SQLPipelineStatement::get_physical_plan
     _physical_plan->set_transaction_context_recursively(_transaction_context);
   }
 
-  // Cache newly created plan for the according sql statement (only if not already cached)
-  if (pqp_cache && !_metrics->query_plan_cache_hit && _translation_info.cacheable) {
+  // Cache newly created plan for the according sql statement (only if not already cached).
+  if (pqp_cache && !_metrics->query_plan_cache_hit && _translation_info.cacheable &&
+      _optimization_context->is_cacheable()) {
```

Comment:
The problem here is that we _could_ have the case where we obtain the LQP from a cache, but have to create the PQP (e.g., because of different cache sizes), right? In that case, the `_optimization_context` member is not set. However, the LQP was cached, so `is_cacheable()` must have been true.